### PR TITLE
Hashicorp.Vagrant version 2.3.3

### PR DESCRIPTION
**Pull request by dfoulkes**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/microsoft/winget-pkgs/pulls) for the same manifest update/change?
- [x] Have you [validated](https://github.com/microsoft/winget-pkgs/blob/master/AUTHORING_MANIFESTS.md#validation) your manifest locally with `winget validate --manifest <path>`?
- [x] Have you tested your manifest locally with `winget install --manifest <path>`?
- [x] Does your manifest conform to the [1.2 schema](https://github.com/microsoft/winget-pkgs/tree/master/doc/manifest/schema/1.2.0)?

Note: `<path>` is the name of the directory containing the manifest you're submitting.

-----

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/winget-pkgs/pull/92840)

### CODE REVIEW

**Comment by dfoulkes** [comment]
@mdanish-kh would you be able to help? I'm back filling a couple of the missed releases on this repository. 
Watching the speed you got the last one out inspired me :-)

File: manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+- silent
+- silentWithProgress
+InstallerSwitches:
+  Custom: /norestart
+InstallerSuccessCodes:
+- 3010
+- 1706
+UpgradeBehavior: install
+InstallLocationRequired: true
+RequireExplicitUpgrade: true
```
Comment (mdanish-kh) [inline manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml:20]:
```suggestion
```
This field is set for packages that auto-update themselves. Setting it to `true` excludes it from being upgraded through `winget upgrade --all`. Is it the case with this package?

File: manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+- interactive
+- silent
+- silentWithProgress
+InstallerSwitches:
+  Custom: /norestart
+InstallerSuccessCodes:
+- 3010
+- 1706
+UpgradeBehavior: install
+InstallLocationRequired: true
```
Comment (mdanish-kh) [inline manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml:19]:
```suggestion
```
This field is part of the schema but not yet implemented.

File: manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+RequireExplicitUpgrade: true
+DisplayInstallWarnings: false
```
Comment (mdanish-kh) [inline manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml:21]:
```suggestion
```
This field is part of the schema but not yet implemented.

File: manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  ProductCode: '{608AAE8F-FAA0-4181-AB93-EA9AAEE0ED6E}'
+- InstallerLocale: en-US
+  Architecture: x86
+  InstallerType: wix
+  InstallerUrl: https://releases.hashicorp.com/vagrant/2.3.3/vagrant_2.3.3_windows_i686.msi
+  InstallerSha256: 2C9AB69BEE74CF19894B40BC7E2EC30B3C00DD9B425C611EB53987D900FC0D5D
+  ProductCode: '{9423FCE4-520A-434F-8771-B373503E44DA}'
+ManifestType: installer
+ManifestVersion: 1.2.0
+ReleaseDate: 2022-08-05
```
Comment (mdanish-kh) [inline manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml:37]:
```suggestion
ReleaseDate: 2022-11-15
```
https://github.com/hashicorp/vagrant/blob/v2.3.3/CHANGELOG.md shows version was released on `November 15, 2022`.

File: manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.locale.en-US.yaml [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+PackageUrl: https://www.vagrantup.com
+License: MIT
+LicenseUrl: https://github.com/hashicorp/vagrant/blob/main/LICENSE
+Copyright: Copyright (c) 2010-2022 Mitchell Hashimoto
+ShortDescription: Vagrant is a tool for building and distributing development environments.
+Tags:
+- developing
+- development
+- virtualization
+ReleaseNotes: 'IMPROVEMENTS: /'
```
Comment (mdanish-kh) [inline manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.locale.en-US.yaml:22]:
```suggestion
ReleaseNotes: |-
  2.3.3 (November 15, 2022)

  IMPROVEMENTS:
  - core: Bump net-ssh dependency to 7.0 and remove patches [GH-12979]
  - synced_folders/rsync: Include ssh extra_args value [GH-12973]

  BUG FIXES:
  - command/serve: Force root level namespace for Google constant [GH-12989]
  - guest/solaris: Fix insecure key authorized keys removal [GH-12740]
  - provider/virtualbox: Fix :private_network support for VirtualBox 7 on macOS [GH-12983]
  - provider/virtualbox: Prevent localization of command output [GH-12994]
  - provisioner/ansible: Update setup packages in debian capability [GH-12832]
```

File: manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+InstallerSuccessCodes:
+- 3010
+- 1706
+UpgradeBehavior: install
+InstallLocationRequired: true
+RequireExplicitUpgrade: true
+DisplayInstallWarnings: false
+Installers:
+- InstallerLocale: en-US
+  Architecture: x64
```
Comment (mdanish-kh) [inline manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml:24]:
```suggestion
- Architecture: x64
```
Is the installer specific to only `en-US` locale? Previous manifest's metadata show that maybe this is not required.

File: manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+  InstallerType: wix
+  InstallerUrl: https://releases.hashicorp.com/vagrant/2.3.3/vagrant_2.3.3_windows_amd64.msi
+  InstallerSha256: DE268C1D580066E23F6E334E3EA14850CD4AEB186E545AF08B1F7217FF67E6E7
+  ProductCode: '{608AAE8F-FAA0-4181-AB93-EA9AAEE0ED6E}'
+- InstallerLocale: en-US
+  Architecture: x86
```
Comment (mdanish-kh) [inline manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml:30]:
```suggestion
- Architecture: x86
```
Is the installer specific to only `en-US` locale? Previous manifest's metadata show that maybe this is not required.

**Review by mdanish-kh — CHANGES_REQUESTED** [review]

File: manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml (hunk 1 above)
Comment (dfoulkes) [reply manifests/h/Hashicorp/Vagrant/2.3.3/Hashicorp.Vagrant.installer.yaml:20]: No that's a mistake

**Review by dfoulkes — COMMENTED** [review]
Reviewed  @mdanish-kh suggestions and implemented.

**Review by mdanish-kh — APPROVED** [review]

**Pull request merged**